connector/ldap/ldap.go

```diff
@@ -165,6 +181,25 @@ func parseScope(s string) (int, bool) {
 	return 0, false
 }
 
+var userMatchers []UserMatcher
```

Comment:
Instead of making sure that this global is set properly in all the right code paths (I think this is what the calls sprinkled on the code below are for?) -- Can we make this a function that reads both config locations and just *returns* `[]UserMatcher`? Like

```go
func (c *ldapConnector) userMatchers() []UserMatcher {
	if len(c.GroupSearch.UserMatchers) > 0 && c.GroupSearch.UserMatchers[0].UserAttr != "" {
		return c.GroupSearch.UserMatchers[:]
	} else {
		return []UserMatcher{
			{
				UserAttr:  c.GroupSearch.UserAttr,
				GroupAttr: c.GroupSearch.GroupAttr,
			},
		}
	}
}
```
and use *that function* where we use `userMatchers` currently?

(I don't believe performance is an issue here.)